src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1216,24 +1271,21 @@ impl LinkCollector<'_, '_> {
                             // FIXME: really it should be `resolution_failure` that does this, not `resolve_with_disambiguator`
                             // See https://github.com/rust-lang/rust/pull/76955#discussion_r493953382 for a good approach
                             for &new_ns in &[other_ns, MacroNS] {
-                                if let Some(res) = self.check_full_res(
-                                    new_ns,
-                                    path_str,
-                                    base_node,
-                                    &extra_fragment,
-                                ) {
+                                if let Some(res) =
+                                    self.check_full_res(new_ns, path_str, base_node, extra_fragment)
+                                {
                                     kind = ResolutionFailure::WrongNamespace(res, ns);
                                     break;
                                 }
                             }
                         }
                         resolution_failure(
                             self,
-                            &item,
+                            diag.item,
```

Comment:
Why not just change these 4 arguments to take diag?

Reply:
Either it was forgotten, or `resolution_failure` is called in a lot of places where creating DiagInfo would have been a chore. I don't recall exactly.

Reply:
Yes, a few other places but they could be made to use the struct.